protocols/kad/src/handler.rs

```diff
@@ -789,7 +790,25 @@ where
             ConnectionEvent::AddressChange(_)
             | ConnectionEvent::ListenUpgradeError(_)
             | ConnectionEvent::LocalProtocolsChange(_) => {}
-            ConnectionEvent::RemoteProtocolsChange(_) => {}
+            ConnectionEvent::RemoteProtocolsChange(ProtocolsChange { protocols }) => {
+                // TODO: We should cache this / it will get simpler with #2831.
+                let kademlia_protocols = self
+                    .config
+                    .protocol_config
+                    .protocol_names()
+                    .iter()
+                    .filter_map(|b| String::from_utf8(b.to_vec()).ok())
+                    .collect::<Vec<_>>();
+
+                let remote_supports_our_kademlia_protocols =
+                    kademlia_protocols.iter().all(|p| protocols.contains(p));
```

Comment:
`any` instead of `all` would suffice, right? In other words, in case the remote supports any of our Kademlia protocols, it is worth adding them to our routing table.